lib/ruby_lsp/ruby_lsp_rails/code_lens.rb

```diff
@@ -0,0 +1,138 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Rails
+    # ![CodeLens demo](../../code_lens.gif)
+    #
+    # This feature adds several CodeLens features for Rails applications using the built-in test framework:
+    # - Run tests in the VS Terminal
+    # - Run tests in the VS Code Test Explorer
+    # - Debug tests
+    #
+    # The
+    # [code lens](https://microsoft.github.io/language-server-protocol/specification#textDocument_codeLens)
+    # request informs the editor of runnable commands such as tests
+    #
+    # Example:
+    #
+    # For the following code, Code Lenses will be added above the class definition above each test method.
+    #
+    # ```ruby
+    # Run
+    # class HelloTest < ActiveSupport::TestCase
+    #   test "outputs hello" do
+    #     # ...
+    #   end
+    #
+    #   test "outputs goodbye" do
+    #     # ...
+    #   end
+    # end
+    # ````
+    #
+    # The code lenses will be displayed above the class and above each test method.
+    class CodeLens < ::RubyLsp::Listener
+      extend T::Sig
+      extend T::Generic
+
+      ResponseType = type_member { { fixed: T::Array[::RubyLsp::Interface::CodeLens] } }
+      BASE_COMMAND = "bin/rails test"
+
+      ::RubyLsp::Requests::CodeLens.add_listener(self)
+
+      sig { override.returns(ResponseType) }
+      attr_reader :response
+
+      sig { params(uri: String, emitter: EventEmitter, message_queue: Thread::Queue).void }
+      def initialize(uri, emitter, message_queue)
+        @response = T.let([], ResponseType)
+        @visibility = T.let("public", String)
+        @prev_visibility = T.let("public", String)
```

Comment:
Sorry if I'm missing something here -- do these values ever change? And does `@prev_visibility` ever get used?

Reply:
That may have been accidentally removed in a refactoring, but probably means we are missing a test, I'll look into.